lib/apiservers/engine/backends/container.go

```diff
@@ -1115,7 +1114,46 @@ func (c *Container) ContainerAttach(name string, ca *backend.ContainerAttachConf
 		}
 	}
 
-	return attachStreams(context.Background(), vc, clStdin, clStdout, clStderr, ca)
+	err = attachStreams(context.Background(), vc, clStdin, clStdout, clStderr, ca)
+	if err != nil {
+		if _, ok := err.(DetachError); ok {
+			log.Infof("Detach detected, tearing down connection")
+			client = c.containerProxy.Client()
+			handle, err = c.Handle(id, name)
+			if err != nil {
+				return err
+			}
+
+			unbind, err := client.Interaction.InteractionUnbind(interaction.NewInteractionUnbindParamsWithContext(ctx).
+				WithConfig(&models.InteractionUnbindConfig{
+					Handle: handle,
+				}))
+			if err != nil {
+				return InternalServerError(err.Error())
+			}
+
+			handle, ok = unbind.Payload.Handle.(string)
```

Comment:
Remembered now

```
cannot use handle (type interface {}) as type string in argument to containers.NewCommitParamsWithContext(ctx).WithHandle: need type assertion
```

Commit expects them as strings, we need to change all portlayer handle usage to interface{} but that's another day's work :)